mailbox/lucene/src/test/java/org/apache/james/mailbox/lucene/search/LuceneMailboxMessageSortOnSentDateTest.java

```diff
@@ -0,0 +1,218 @@
+
+        uid1 = MessageUid.of(1);
+        uid2 = MessageUid.of(2);
+
+        id1 = factory.generate();
+        id2 = factory.generate();
+
+        GregorianCalendar cal = new GregorianCalendar();
+
+        cal.set(2017, 0, 10);
+        smallestDate = cal.getTime();
+
+        cal.set(2017, 0, 12);
+        biggestDate = cal.getTime();
+    }
+
+    private void indexMailWithSentDateAndInternalDate(MessageId id, MessageUid uid, Date internalDate, String sentDate) throws Exception {
+        Map<String, String> header = new HashMap<String, String>();
+        if (sentDate != null) {
+            header.put("Date", sentDate);
+        }
+
+        SimpleMailboxMembership membership = new SimpleMailboxMembership(
+            id,
+            mailbox.getMailboxId(),
+            uid,
+            0,
+            internalDate,
+            20,
+            new Flags(Flag.DELETED),
+            "My Otherbody2".getBytes(),
+            header);
+        index.add(session, mailbox, membership);
+    }
+
+    @Test
+    public void sortOnSentDateShouldSortUsingSendDateIfPresent() throws Exception {
+        indexMailWithSentDateAndInternalDate(id1, uid1, biggestDate, "Date: Fri, 2 Jun 2017 10:37:47 -0400");
+        indexMailWithSentDateAndInternalDate(id2, uid2, smallestDate, "Date: Fri, 2 Jun 2017 11:37:47 -0400");
+
+        SearchQuery query = new SearchQuery();
+        query.andCriteria(SearchQuery.all());
+        query.setSorts(Arrays.asList(new SearchQuery.Sort(SortClause.SentDate, false)));
+        Iterator<MessageUid> result = index.search(session, mailbox, query);
+
+        assertThat(result).containsExactly(uid1, uid2);
+    }
+
+    @Test
+    public void sortOnSentDateShouldSortUsingInternalDateIfNoSentDate() throws Exception {
+        indexMailWithSentDateAndInternalDate(id1, uid1, smallestDate, null);
+        indexMailWithSentDateAndInternalDate(id2, uid2, biggestDate, null);
+
+        SearchQuery query = new SearchQuery();
+        query.andCriteria(SearchQuery.all());
+        query.setSorts(Arrays.asList(new SearchQuery.Sort(SortClause.SentDate, false)));
+        Iterator<MessageUid> result = index.search(session, mailbox, query);
+
+        assertThat(result).containsExactly(uid1, uid2);
+    }
+
+    @Test
+    public void sortOnSentDateShouldCompareInternalDateAgainstSentDateIfOneOfTheMessageHasNoSentDate() throws Exception {
+        indexMailWithSentDateAndInternalDate(id1, uid1, biggestDate, null);
+        indexMailWithSentDateAndInternalDate(id2, uid2, smallestDate, "Date: Fri, 2 Jun 2017 11:37:47 -0400");
+
+        SearchQuery query = new SearchQuery();
+        query.andCriteria(SearchQuery.all());
+        query.setSorts(Arrays.asList(new SearchQuery.Sort(SortClause.SentDate, false)));
+        Iterator<MessageUid> result = index.search(session, mailbox, query);
+
+        assertThat(result).containsExactly(uid1, uid2);
+    }
+
+    private final class SimpleMailbox implements Mailbox {
```

Comment:
Why do we need yet an other override? Can't you work with a mock or with the *real* simple mailbox?